qiskit/circuit/quantumcircuit.py

```diff
@@ -1151,7 +1151,10 @@ def add_register(self, *regs):
                 regs = (QuantumRegister(regs[0], "q"),)
             elif len(regs) == 2 and all(isinstance(reg, int) for reg in regs):
                 # QuantumCircuit with anonymous wires e.g. QuantumCircuit(2, 3)
-                regs = (QuantumRegister(regs[0], "q"), ClassicalRegister(regs[1], "c"))
+                if regs[1] == 0:
+                    regs = (QuantumRegister(regs[0], "q"),)
+                else:
+                    regs = (QuantumRegister(regs[0], "q"), ClassicalRegister(regs[1], "c"))
```

Comment:
If size 0 is allowed then this modification is not needed i think

Reply:
@ajavadia Removing this `if` statement triggers some issues with the tests. There is a difference between a circuit with no classical registers (i.e. an empty list), or a circuit with a classical register of size 0.

![image](https://user-images.githubusercontent.com/883786/122742366-27327180-d286-11eb-9a87-93f9f56329a1.png)

The behaviour in this PR. e.g. `QuantumCircuit(2,0)` creates a circuit without classical registers, is consistent with current behaviour (and tests).